lib/active_model/serializer/attributes.rb

```diff
@@ -1,55 +1,23 @@
 module ActiveModel
   class Serializer
     module Attributes
-      # @api private
-      class Attribute
-        delegate :call, to: :reader
-
-        attr_reader :name, :reader
-
-        def initialize(name)
-          @name = name
-          @reader = :no_reader
-        end
-
-        def self.build(name, block)
-          if block
-            AttributeBlock.new(name, block)
-          else
-            AttributeReader.new(name)
-          end
-        end
-      end
-      # @api private
-      class AttributeReader < Attribute
-        def initialize(name)
-          super(name)
-          @reader = ->(instance) { instance.read_attribute_for_serialization(name) }
-        end
-      end
-      # @api private
-      class AttributeBlock < Attribute
-        def initialize(name, block)
-          super(name)
-          @reader = ->(instance) { instance.instance_eval(&block) }
-        end
-      end
-
       extend ActiveSupport::Concern
 
       included do
         with_options instance_writer: false, instance_reader: false do |serializer|
-          serializer.class_attribute :_attribute_mappings # @api private : maps attribute key names to names to names of implementing methods, @see #attribute
-          self._attribute_mappings ||= {}
+          serializer.class_attribute :_attributes_data # @api private
+          self._attributes_data ||= {}
```

Comment:
Aaand I won't, because you sold me on removing the redundant `key` from the `Attribute` structure. Which leaves, in this PR, the renaming from `attribute_mappings` to `attributes_data`, and the actual modification of your `Attribute` class. I don't feel strongly about the former, although I do not like the name `mapping` when really the structure holds information abous the attribute (in a philosophical sense). The latter though, I believe, adds clarity (much shorter) and simplicity (avoids storing lambdas calling blocks). Thoughts?